### PR TITLE
[plugin.video.retrospect] v5.4.0+matrix.1

### PR DESCRIPTION
**Pull request by basrieter**
### Description

This update fixes some channel issues and contains a simplification in the 'playing back of items' code that was long over due. 

There are many changed files, but only minor changes.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.retrospect/channels/channel.streams/radio538/chn_radio538.py [hunk 1]
```diff
@@ -203,112 +281,154 @@ def create_live_channel(self, result_set):
...
-                stream = "https:%s" % (stream,)
-            part = item.create_new_empty_media_part()
-            part.Name = title_format % (hour, start_time_stamp.tm_min)
-            part.append_media_stream(stream, 0)
-            hour += 1
-        else:
-            Logger.warning("Found item without streams: %s", item)
-            return None
-        return item
+    # No longer used
```
Comment (enen92) [inline plugin.video.retrospect/channels/channel.streams/radio538/chn_radio538.py:311]: Maybe remove the comment block

File: plugin.video.retrospect/channels/channel.videos/twit/chn_twit.py [hunk 2]
```diff
@@ -74,62 +86,11 @@ def add_live_stream(self, data):
...
-
-        for bitrate in media_urls:
-            playback_part.append_media_stream(media_urls[bitrate], bitrate)
-
-        Logger.debug("Streams: %s", playback_part)
-        playback_item.complete = True
-        item.items.append(playback_item)
-        Logger.debug("Appended: %s", playback_item)
+        item = MediaItem("\a.: TWiT.TV Live :.",
+                         "http://iphone-streaming.ustream.tv/uhls/1524/streams/live/iphone/playlist.m3u8",
```
Comment (enen92) [inline plugin.video.retrospect/channels/channel.videos/twit/chn_twit.py:90]: Odd stream URL this one

File: plugin.video.retrospect/channels/channel.streams/radio538/chn_radio538.py (hunk 1 above)
Comment (basrieter) [reply plugin.video.retrospect/channels/channel.streams/radio538/chn_radio538.py:311]: It's still a TODO for the next version, but I did not want to wait that long.

**Review by enen92 — APPROVED** [review]

File: plugin.video.retrospect/channels/channel.videos/twit/chn_twit.py (hunk 2 above)
Comment (basrieter) [reply plugin.video.retrospect/channels/channel.videos/twit/chn_twit.py:90]: Indeed, but an official stream: https://wiki.twit.tv/wiki/TWiT_Live

**Pull request merged**